File: frontend/src/api/communityDashboard.ts
import apiClient from './index'

// ===== Response Types (mirrors backend schemas/community_dashboard.py) =====

export interface CommunityMetrics {
  total_contents: number
  published_contents: number
  reviewing_contents: number
  draft_contents: number
  total_committees: number
  total_members: number
  upcoming_meetings: number
  active_channels: number
  active_campaigns: number
}

export interface MonthlyTrend {
  month: string  // "2025-08"
  count: number
}

export interface ChannelStats {
  wechat: number
  hugo: number
  csdn: number
  zhihu: number
}

export interface RecentContentItem {
  id: number
  title: string
  status: string
  work_status: string | null
  created_at: string
  owner_name: string
}

export interface UpcomingMeetingItem {
  id: number
  title: string
  scheduled_at: string
  committee_name: string
  status: string
}

export interface RecentCampaignItem {
  id: number
  name: string
  type: string
  status: string
  start_date: string | null
  end_date: string | null
  owner_name: string | null
}

export interface CalendarEvent {
  id: string
  type: string  // "meeting" | "publish" | "member_join"
  title: string
  date: string
  color: string
  resource_id: number
  resource_type: string
}

export interface CommunityDashboardResponse {
  metrics: CommunityMetrics
  monthly_trend: MonthlyTrend[]
  channel_stats: ChannelStats
  recent_contents: RecentContentItem[]
  upcoming_meetings: UpcomingMeetingItem[]
  recent_campaigns: RecentCampaignItem[]
  calendar_events: CalendarEvent[]
}

// ===== Superuser Overview =====

export interface CommunityOverviewItem {
  id: number
  name: string
  slug: string
  is_active: boolean
  member_count: number
  total_contents: number
  published_contents: number
  reviewing_contents: number
  committee_count: number
  upcoming_meeting_count: number
  active_channel_count: number
}

export interface SuperuserOverviewResponse {
  total_communities: number
  communities: CommunityOverviewItem[]
}

// ===== API Functions =====

/** 获取社区工作台聚合数据 */
export async function getCommunityDashboard(
  communityId: number
): Promise<CommunityDashboardResponse> {
  const { data } = await apiClient.get<CommunityDashboardResponse>(
    `/communities/${communityId}/dashboard`,
    { headers: { 'X-Community-Id': String(communityId) } }
  )
  return data
}

/** 超管全局社区总览（批量统计，消除 N+1） */
export async function getSuperuserOverview(): Promise<SuperuserOverviewResponse> {
  const { data } = await apiClient.get<SuperuserOverviewResponse>(
    '/communities/overview/stats'
  )
  return data
}

File: backend/app/schemas/community_dashboard.py
"""
社区工作台（Community Dashboard）Schemas

用于 GET /communities/{id}/dashboard 聚合 API 的请求/响应数据结构。
"""

from datetime import datetime
from typing import Dict, List, Optional

from pydantic import BaseModel


# ── 指标卡片 ──────────────────────────────────────────────────────────

class CommunityMetrics(BaseModel):
    """工作台 8 项核心指标"""
    # 内容指标
    total_contents: int = 0
    published_contents: int = 0
    pending_review_contents: int = 0
    draft_contents: int = 0
    # 治理指标
    committees_count: int = 0
    members_count: int = 0
    upcoming_meetings_count: int = 0
    active_channels_count: int = 0


# ── 趋势图 ──────────────────────────────────────────────────────────

class MonthlyTrend(BaseModel):
    """单月发布数据点"""
    month: str  # "2025-08" 格式
    count: int


class ChannelStats(BaseModel):
    """各渠道发布数量"""
    wechat: int = 0
    hugo: int = 0
    csdn: int = 0
    zhihu: int = 0


# ── 最近内容 ──────────────────────────────────────────────────────────

class RecentContentItem(BaseModel):
    """工作台内容列表项"""
    id: int
    title: str
    status: str          # draft / reviewing / approved / published
    work_status: str     # planning / in_progress / completed
    created_at: datetime
    owner_name: Optional[str] = None

    model_config = {"from_attributes": True}


# ── 即将召开的会议 ────────────────────────────────────────────────────

class UpcomingMeetingItem(BaseModel):
    """工作台会议列表项"""
    id: int
    title: str
    scheduled_at: datetime
    committee_name: str
    status: str          # scheduled / in_progress / completed / cancelled

    model_config = {"from_attributes": True}


# ── 日历事件 ────────────────────────────────────────────────────────

class CalendarEvent(BaseModel):
    """日历视图事件"""
    id: int
    type: str            # meeting / publish / member_join
    title: str
    date: datetime
    color: str           # 前端渲染色值
    resource_id: int     # 对应资源的 ID（会议 ID / 内容 ID / 用户 ID）
    resource_type: str   # 跳转时使用的路由类型


# ── 聚合响应 ────────────────────────────────────────────────────────

class CommunityDashboardResponse(BaseModel):
    """
    社区工作台完整聚合响应。
    前端通过单次请求获取工作台所需的全部数据。
    """
    community_id: int
    community_name: str
    community_logo: Optional[str] = None

    # 指标卡片
    metrics: CommunityMetrics

    # 趋势图：近 6 个月按月聚合
    publish_trend: List[MonthlyTrend] = []

    # 渠道统计
    channel_stats: ChannelStats = ChannelStats()

    # 工作队列
    recent_contents: List[RecentContentItem] = []
    upcoming_meetings: List[UpcomingMeetingItem] = []

    # 日历事件（近 30 天 + 未来 60 天）
    calendar_events: List[CalendarEvent] = []


# ── 超管社区总览 ────────────────────────────────────────────────────

class CommunityOverviewItem(BaseModel):
    """超管总览中单个社区的摘要信息"""
    id: int
    name: str
    slug: str
    logo_url: Optional[str] = None
    is_active: bool
    members_count: int = 0
    contents_count: int = 0
    published_count: int = 0
    pending_review_count: int = 0
    committees_count: int = 0
    upcoming_meetings_count: int = 0
    active_channels_count: int = 0
    last_activity_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class SuperuserOverviewResponse(BaseModel):
    """超管社区总览聚合响应（解决 N+1 查询）"""
    total_communities: int
    total_members: int
    total_contents: int
    communities: List[CommunityOverviewItem] = []
